src/main/scala/edu/ie3/simona/model/participant/hp/HpModel.scala

```diff
@@ -246,19 +263,23 @@ class HpModel private (
 
     val demandHouse = thermalDemands.houseDemand
     val demandHeatStorage = thermalDemands.heatStorageDemand
+    val demandDomesticHotWaterStorage =
+      thermalDemands.domesticHotWaterStorageDemand
     val noHeatStorageOrEmpty = thermalGridState.isHeatStorageEmpty
 
     val turnHpOn =
       (demandHouse.hasRequiredDemand && noHeatStorageOrEmpty) ||
         (demandHouse.hasPossibleDemand && wasRunningLastPeriod ||
           demandHeatStorage.hasRequiredDemand ||
-          (demandHeatStorage.hasPossibleDemand && wasRunningLastPeriod))
+          (demandHeatStorage.hasPossibleDemand && wasRunningLastPeriod)) ||
+        demandDomesticHotWaterStorage.hasRequiredDemand
```

Comment:
DId you forget the check for `demandDomesticHotWaterStorage.hasPossibleDemand && wasRunningLastPeriod`?
Also, why is this check no inside the previous bracket, while the normal heat storage is?

Reply:
What do you mean be the check for `demandDomesticHotWaterStorage.hasPossibleDemand && wasRunningLastPeriod`?

But one of the brackets was wrong. Good catch!